lib/Conversion/FIRRTLToHW/LowerToHW.cpp

```diff
@@ -1004,7 +1004,7 @@ void FIRRTLModuleLowering::lowerModuleBody(
     // anything outside the module.  Inputs are lowered to zero.
     if (isZeroWidth && port.isInput()) {
       Value newArg = bodyBuilder.create<WireOp>(
-          port.type, "." + port.getName().str() + ".0width_input");
+          port.type, "__" + port.getName().str() + "_0width_input");
       oldArg.replaceAllUsesWith(newArg);
```

Comment:
Agreed that the `.` might be a bit weird. But shouldn't _any_ choice of name we put in here eventually lead to a properly sanitized output in `ExportVerilog`? This seems to point more to a bug in name legalization than a poor choice of naming :thinking:

Reply:
Yes, the problem isn't the "." as that gets fixed at emission time.  The problem is that it isn't being fixed at emission time in this one case.

Reply:
I think you could make a test case with a name conflict using either this or the old way which would still result in the incorrect output (but due to renaming, rather than not getting the canonicalized name)